src/interactions/dragInteraction.ts

```diff
@@ -0,0 +1,134 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Interaction {
+  export class Drag extends AbstractInteraction {
+    private _dragging = false;
+    private _mouseDispatcher: Plottable.Dispatcher.Mouse;
+    private _dragOrigin: Point;
+    private _dragStartCallback: (p: Point) => any;
+    private _dragCallback: (start: Point, end: Point) => any;
+    private _dragEndCallback: (start: Point, end: Point) => any;
+
+    public _anchor(component: Component.AbstractComponent, hitBox: D3.Selection) {
+      super._anchor(component, hitBox);
+      this._mouseDispatcher = Dispatcher.Mouse.getDispatcher(
+                                <SVGElement> (<any> this._componentToListenTo)._element.node()
+                              );
+      this._mouseDispatcher.onMouseDown("Interaction.Drag" + this.getID(),
+        (p: Point, e: MouseEvent) => this._startDrag(p, e));
+      this._mouseDispatcher.onMouseMove("Interaction.Drag" + this.getID(),
+        (p: Point, e: MouseEvent) => this._doDrag(p, e));
+      this._mouseDispatcher.onMouseUp("Interaction.Drag" + this.getID(),
+        (p: Point, e: MouseEvent) => this._endDrag(p, e));
+    }
+
+    private _translateAndConstrain(p: Point) {
+      var translatedP = this._translateToComponentSpace(p);
+      var constrainedX = Math.min(Math.max(0, translatedP.x), this._componentToListenTo.width());
+      var constrainedY = Math.min(Math.max(0, translatedP.y), this._componentToListenTo.height());
```

Comment:
Did I place a clamp call in utils at this point?  If not, we might want to move this to a Util as this may be used elsewhere

As a reference, I wrote the util as
clamp(value, a, b)
  var minValue = min(a, b)
  var maxValue = max(a, b)
  if value is less than min, return min
  if value is greater than max, return max
  return value

The way you have it also I think should work / is clean